Thermal brems bugfixes

settwi opened the pull request:
Fixes #229 for legacy and dev't sections of code

samaloney commented on sunkit_spex/legacy/thermal.py:
```diff
@@ -764,15 +768,15 @@ def _warn_if_input_outside_valid_range(input_values, grid_range, param_name, par
 
 def _calculate_abundances(abundance_type, relative_abundances):
-    abundances = DEFAULT_ABUNDANCES[abundance_type].data
+    abundances = fetch_default_abundances(abundance_type)
```
What about then we don't need a new function or to alter the names?
```suggestion
    abundances = DEFAULT_ABUNDANCES(abundance_type).data.copy()
```

settwi replied:
sure we could do a copy. the reasoning was, then whenever you reference DEFAULT_ABUNDANCES, you need to remember to do the .copy. with the function it does it by default

samaloney reviewed (changes requested):
Good catch! What would you think about keeping original name and just do a `.copy()` inside the functions?

Also might be good to move the removal of the other code to a different PR - I'm not sure we ever fully decided on which approach to use and why so should make it easier to find.

Also a test for the new code to make sure this doesn't happen again would be good.

settwi committed:
legacy: abundance bugfix and test

settwi committed:
dev: thermal patches (see more)

- ensure DEFAULT_ABUNDANCES table is not modified during model evaluation
- remove extraneous (or incorrect) unit conversions
- add test for DEFAULT_ABUNDANCES table correctness

settwi committed:
changelog

KriSun95 reviewed (approved):
LGTM
